src/ui/src/core_components/input/CoreColorInput.vue

```diff
@@ -0,0 +1,100 @@
+<template>
+	<BaseInputWrapper
+		ref="rootInstance"
+		:label="fields.label.value"
+		class="CoreColorInput"
+	>
+		<BaseInputColor
+			:value="formValue"
+			:custom-colors="colorList"
+			@update:value="handleInput($event, 'wf-change')"
+			@change="handleInput($event, 'wf-change-finish')"
+		/>
+	</BaseInputWrapper>
+</template>
+
+<script lang="ts">
+import { ComponentPublicInstance } from "vue";
+import { cssClasses } from "../../renderer/sharedStyleFields";
+import { FieldType, WriterComponentDefinition } from "../../writerTypes";
+import BaseInputColor from "../base/BaseInputColor.vue";
+import BaseInputWrapper from "../base/BaseInputWrapper.vue";
+
+const description =
+	"A user input component that allows users to select a color using a color picker interface.";
+
+const onChangeHandlerStub = `
+def onchange_handler(state, payload):
+
+	# Set the state variable "new_color" to the new value, provided as string.
+
+	state["new_color"] = payload`;
+
+const definition = {
+	name: "Color Input",
+	description,
+	category: "Input",
+	fields: {
+		label: {
+			name: "Label",
+			init: "Input Label",
+			type: FieldType.Text,
+		},
+		colorList: {
+			name: "Color List",
+			desc: "List of predefined colors",
```

Comment:
`list of predefined colors in hexadecimal format`

Reply:
what is the benefits of the event `wf-change-finish` ? I feel like the `wf-change` event covers all cases.

![image](https://github.com/user-attachments/assets/d54719f4-7f58-4f2b-85f4-79e93ba9fd1b)

Reply:
@FabienArcellier I had the same question when seing this event on `CoreTextInput` ([here](https://github.com/writer/writer-framework/blob/6f7a03c8dce216adfe27bb0809e68045e0ed2f4b/src/ui/src/core_components/input/CoreTextInput.vue#L74-L82)). The difference is the HTML event tracked

https://github.com/writer/writer-framework/blob/bb8dc80ab1d552cfc3ff57957da27d27fcef2be3/src/ui/src/core_components/base/BaseInputColor.vue#L8-L9

- `@input` / `wf-change` is triggered when the value change in real time (like user type a letter or slide the color picker)
- `@change` / `wf-change-finish` is trigger when user finish to write and click outside (lose focus)

Thus it's better to have both because there are some situations where you want to execute the action once the user finished his changes